src/lib/updating-element.ts

```diff
@@ -495,14 +494,15 @@ export abstract class UpdatingElement extends HTMLElement {
   }
 
   connectedCallback() {
-    this._updateState = this._updateState | STATE_HAS_CONNECTED;
     // Ensure first connection completes an update. Updates cannot complete
-    // before connection and if one is pending connection the
-    // `_hasConnectionResolver` will exist. If so, resolve it to complete the
-    // update, otherwise requestUpdate.
-    if (this._hasConnectedResolver) {
-      this._hasConnectedResolver();
-      this._hasConnectedResolver = undefined;
+    // before connection.
+    this.enableUpdating();
+  }
+
+  protected enableUpdating() {
```

Comment:
LGTM on this part of the PR. I wouldn't mind this PR split into two for review purposes.

Reply:
Agree would have been better that way. If there's additional feedback on the new decorator, will do.